joomla/joomla-cms
# [6.2] Ensure isMediaFile defaults to all supported formats when mediatypes is not specified

### Summary of Changes

- If mediatypes is not specified in the input, fallback to an explodable string of all supported formats (0,1,2,3) rather than images only (0).

### Testing Instructions

1. Download the following file: [Example MP4](https://file-examples.com/wp-content/storage/2017/04/file_example_MP4_480_1_5MG.mp4)
2. Upload the video file in to the root of the media library - by default, the path to the file should be: local-images:/file_example_MP4_480_1_5MG.mp4
3. In a custom component or module (e.g. [mod_test_43915.zip](https://github.com/user-attachments/files/16736419/mod_test_43915.zip) created by @degobbis), call:
`\Joomla\Component\Media\Administrator\Model\FileModel->getFileInformation('local-images:/file_example_MP4_480_1_5MG.mp4');`

### Actual result BEFORE applying this Pull Request

With mediatypes not present or empty in the input object, calling FileModel::getFileInformation() with a path to anything other than an image file will generate an InvalidPathException.

The exception is thrown by ApiModel::getFile(), and will also be thrown if this function is called directly with a separate adapter and path.

The exception is triggered due to the fact that ApiModel::isMediaFile() will return false for a non-image file if mediatypes is not present or is empty in the input object. Without mediatypes, allowed formats will be limited to images only, rather than images, audio files, videos and documents.

### Expected result AFTER applying this Pull Request

FileModel::getFileInformation() (and ApiModel::getFile()) will return an object containing file information for image and non-image files, assuming the referenced file exists at the location specified by the path.

(The namespace for all classes mentioned above is: Joomla\Component\Media\Administrator\Model.)

### Link to documentations

No documentation changes for docs.joomla.org needed.
No documentation changes for manual.joomla.org needed.
